lib/Driver/Driver.cpp

```diff
@@ -1368,18 +1368,20 @@ void Driver::buildOutputInfo(const ToolChain &TC, const DerivedArgList &Args,
 
     switch (OutputModeArg->getOption().getID()) {
     case options::OPT_emit_executable:
+      assert(!Args.hasArg(options::OPT_static) &&
+             "-static may not be used with -emit-executable.");
```

Comment:
This shouldn't be an assertion; it should be a proper diagnostic somewhere for a user who doesn't know.